src/Microsoft.AspNetCore.Mvc.Razor/Internal/TagHelperComponentPropertyActivator.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Razor.TagHelpers;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.Razor
+{
+    /// <summary>
+    /// Default implementation of <see cref="ITagHelperComponentPropertyActivator"/>.
+    /// </summary>
+    internal class TagHelperComponentPropertyActivator : ITagHelperComponentPropertyActivator
+    {
+        private readonly ConcurrentDictionary<Type, PropertyActivator<ViewContext>[]> _propertiesToActivate;
+        private readonly Func<Type, PropertyActivator<ViewContext>[]> _getPropertiesToActivate;
+        private static readonly Func<PropertyInfo, PropertyActivator<ViewContext>> _createActivateInfo = CreateActivateInfo;
+
+        public TagHelperComponentPropertyActivator()
+        {
+            _propertiesToActivate = new ConcurrentDictionary<Type, PropertyActivator<ViewContext>[]>();
+            _getPropertiesToActivate = type => PropertyActivator<ViewContext>.GetPropertiesToActivate(
```

Comment:
pull this little piece (the lambda) into a static method just so it's consistent with `CreateActivateInfo`